src/eos-updater-poll-common.c

```diff
@@ -603,79 +376,87 @@ fetch_latest_commit (OstreeRepo *repo,
                                           error))
         return FALSE;
 
-      g_clear_pointer (&checksum, g_free);
-      if (!fetch_commit_checksum (repo,
-                                  cancellable,
-                                  remote_name,
-                                  ref,
-                                  url_override,
-                                  &checksum,
-                                  error))
+      if (!parse_latest_commit (repo, refspec, &redirect_followed, &checksum,
```

Comment:
Aren't we leaking `checksum` here in the case that a redirect is followed?

Reply:
Oops, yes. I’ll push a fixup. Well spotted.